{"version":3,"sources":["../../src/server/dynamic-rendering-utils.ts"],"sourcesContent":["/**\n * This function constructs a promise that will never resolve. This is primarily\n * useful for dynamicIO where we use promise resolution timing to determine which\n * parts of a render can be included in a prerender.\n *\n * @internal\n */\nexport function makeHangingPromise<T>(\n  signal: AbortSignal,\n  expression: string\n): Promise<T> {\n  const hangingPromise = new Promise<T>((_, reject) => {\n    signal.addEventListener(\n      'abort',\n      () => {\n        reject(\n          new Error(\n            `During prerendering, ${expression} rejects when the prerender is complete. Typically these errors are handled by React but if you move ${expression} to a different context by using \\`setTimeout\\`, \\`after\\`, or similar functions you may observe this error and you should handle it in that context.`\n          )\n        )\n      },\n      { once: true }\n    )\n  })\n  // We are fine if no one actually awaits this promise. We shouldn't consider this an unhandled rejection so\n  // we attach a noop catch handler here to suppress this warning. If you actually await somewhere or construct\n  // your own promise out of it you'll need to ensure you handle the error when it rejects.\n  hangingPromise.catch(ignoreReject)\n  return hangingPromise\n}\n\nfunction ignoreReject() {}\n"],"names":["makeHangingPromise","signal","expression","hangingPromise","Promise","_","reject","addEventListener","Error","once","catch","ignoreReject"],"mappings":"AAAA;;;;;;CAMC;;;;+BACeA;;;eAAAA;;;AAAT,SAASA,mBACdC,MAAmB,EACnBC,UAAkB;IAElB,MAAMC,iBAAiB,IAAIC,QAAW,CAACC,GAAGC;QACxCL,OAAOM,gBAAgB,CACrB,SACA;YACED,OACE,IAAIE,MACF,CAAC,qBAAqB,EAAEN,WAAW,qGAAqG,EAAEA,WAAW,qJAAqJ,CAAC;QAGjT,GACA;YAAEO,MAAM;QAAK;IAEjB;IACA,2GAA2G;IAC3G,6GAA6G;IAC7G,yFAAyF;IACzFN,eAAeO,KAAK,CAACC;IACrB,OAAOR;AACT;AAEA,SAASQ,gBAAgB"}